plenum/test/node_catchup/test_req_id_key_error.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+from plenum.test.testable import spyable
+from plenum.test.test_node import TestNode
+from plenum.common.types import f
+from typing import Any
+from plenum.test.test_node import node_spyables
+
+from stp_core.common.log import getlogger
+logger = getlogger()
+
+
+@spyable(methods=node_spyables)
+class TestNodeReqID(TestNode):
+
+    def postTxnFromCatchupAddedToLedger(self, ledger_id: int, txn: Any):
+        rh = self.postRecvTxnFromCatchup(ledger_id, txn)
+        if rh:
+            rh.updateState([txn], isCommitted=True)
+            state = self.getState(ledger_id)
+            state.commit(rootHash=state.headHash)
+        logger.debug("remove req_ID if exists")
+        if txn.get(f.REQ_ID.nm):
+            txn.pop(f.REQ_ID.nm, None)
+        try:
+            self.updateSeqNoMap([txn])
+        except KeyError:
+            logger.info("Got KeyError while transactions catchup")
+            self.stop()
+        self._clear_req_key_for_txn(ledger_id, txn)
+
+@pytest.fixture(scope="module")
+def testNodeClass(patchPluginManager):
+    return TestNodeReqID
+
+def test_req_id_key_error(txnPoolNodeSet, nodeSetWithNodeAddedAfterSomeTxns):
+    # We assume, that pool is consisted of 5 node, 4 node until pool init and 1 added node after.
+    # We assume, that pool wouldn't be broken after getting txns without reqId
+    assert len(txnPoolNodeSet) == 5
```

Comment:
I think this check doesn't show that all 5 nodes are still running. `KeyError` below is caught and silently ignored.